doctor_visits/delphi_doctor_visits/update_sensor.py

```diff
@@ -225,12 +262,92 @@ def update_sensor(
         "include": sensor_include,
     }
 
+    if sensorize:
+
+        # Convert data in dict of dicts format to single pd.DataFrame
+        signal_dfs = []
+        for geo_id in unique_geo_ids:
+            incl = output_dict["include"][geo_id]
+            geo_df = pd.DataFrame(
+                data={"signal":output_dict["rates"][geo_id][incl],
+                      "time": burn_in_dates[final_sensor_idxs][incl]})
+            geo_df["geo"] = geo_id
+            signal_dfs.append(geo_df)
+        signal_df = pd.concat(signal_dfs)
+
+        # Load target (7-day averaged case incidence proportion)
+        target_df = covidcast.signal("jhu-csse",
+                                     "confirmed_7dav_incidence_prop",
+                                     geo_type = geo.lower(),
+                                     start_day = pd.to_datetime(sensor_dates[0]),
+                                     end_day = pd.to_datetime(sensor_dates[-1]))
+        target_df = target_df[["geo_value","time_value","value"]]
+
+        geomapper = GeoMapper()
+        fips_pop = pd.DataFrame({"fips":sorted(list(geomapper.get_geo_values("fips")))})
+        fips_pop = fips_pop[fips_pop.fips.str.slice(start=2) != "000"]
+        fips_pop = geomapper.add_population_column(fips_pop,"fips",geocode_col="fips")
+        if geo.lower() == "state":
+            geo_weights = geomapper.replace_geocode(
+                fips_pop,"fips","state_id",from_col="fips",new_col="geo",date_col=None
+            )
+        elif geo.lower() == "hrr":
+            geo_weights = geomapper.replace_geocode(
+                fips_pop,"fips","hrr",from_col="fips",new_col="geo",date_col=None
+            )
+        elif geo.lower() == "msa":
+            geo_weights = geomapper.replace_geocode(
+                fips_pop,"fips","msa",from_col="fips",new_col="geo",date_col=None
+            )
+        elif geo.lower() == "county":
+            geo_weights = fips_pop.rename(columns={"fips":"geo"})
+        geo_weights = geo_weights.rename(columns={"population":"weight"})
```

Comment:
this feels like something that could eventually live in the geomapper util to get populations for everything, but for now would recommend moving this into its own function that returns the weight DF.